<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QWidget" name="widget" native="true">
        <widget class="QGroupBox" name="grpNetworkConfiguration">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>0</y>
           <width>276</width>
           <height>231</height>
          </rect>
         </property>
         <property name="title">
          <string>Network Configuration</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="7" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Current Class</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinNeurons">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSpinBox" name="spinOutputLayer">
            <property name="maximum">
             <number>6</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Hidden Layers</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spinHiddenLayers">
            <property name="maximum">
             <number>10</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="cmbActivation">
            <item>
             <property name="text">
              <string>SIGMOID</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>TANH</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Class Count</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="spinMaxEpochs">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="value">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Learning Rate</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Task Mode</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="spinLR">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.000100000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.005000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Activation</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Neurons Per Layer</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Max Epochs</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cmbMode">
            <property name="maxVisibleItems">
             <number>10</number>
            </property>
            <item>
             <property name="text">
              <string>Single Layer Classification</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Single Layer Regression</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Multi Layer Classification</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Multi Layer Regression</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QSpinBox" name="spinCurrentClass"/>
          </item>
         </layout>
        </widget>
        <widget class="QGroupBox" name="grpActions">
         <property name="geometry">
          <rect>
           <x>469</x>
           <y>219</y>
           <width>247</width>
           <height>101</height>
          </rect>
         </property>
         <property name="title">
          <string>Actions</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QPushButton" name="btnCreate">
            <property name="text">
             <string>Create Network</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="btnTrain">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Start Training</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btnReset">
            <property name="text">
             <string>Reset Network</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="btnTest">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QGroupBox" name="grpStates">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>320</y>
           <width>301</width>
           <height>201</height>
          </rect>
         </property>
         <property name="title">
          <string>States</string>
         </property>
         <widget class="QLabel" name="lblEpoch">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>30</y>
            <width>241</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Epoch: 0</string>
          </property>
         </widget>
         <widget class="QLabel" name="lblError">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>60</y>
            <width>241</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Error: 0.0000</string>
          </property>
         </widget>
         <widget class="ErrorGraph" name="widgetErrorGraph" native="true">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>80</y>
            <width>281</width>
            <height>111</height>
           </rect>
          </property>
         </widget>
        </widget>
        <widget class="RenderArea" name="renderArea" native="true">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>9</y>
           <width>431</width>
           <height>521</height>
          </rect>
         </property>
         <property name="styleSheet">
          <string notr="true">background-color: rgb(246, 245, 244);</string>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ErrorGraph</class>
   <extends>QWidget</extends>
   <header>errorgraph.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>RenderArea</class>
   <extends>QWidget</extends>
   <header>renderarea.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
